src/vec.ts

```diff
@@ -209,13 +192,14 @@ export namespace vec {
     return {
       mult: (m: Matrix) => matrixBuilder(matrixMult(m, _m)),
       translate: (x: number, y: number) =>
-        matrixBuilder(matrixMult([1, 0, x, 0, 1, y, 0, 0, 1], _m)),
-      rotate: (a: number) =>
-        matrixBuilder(
-          matrixMult([Math.cos(a), -Math.sin(a), 0, Math.sin(a), Math.cos(a), 0, 0, 0, 1], _m)
-        ),
-      scale: (x: number, y: number) => matrixBuilder(matrixMult([x, 0, 0, 0, y, 0, 0, 0, 1], _m)),
-      shear: (x: number, y: number) => matrixBuilder(matrixMult([1, x, 0, y, 1, 0, 0, 0, 1], _m)),
+        matrixBuilder(matrixMult(matrixCreate(1, 0, 0, 1, x, y), _m)),
+      rotate: (a: number) => {
+        const c = Math.cos(a)
+        const s = Math.sin(a)
+        return matrixBuilder(matrixMult(matrixCreate(c, s, -s, c), _m))
+      },
+      scale: (x: number, y: number) => matrixBuilder(matrixMult(matrixCreate(x, 0, 0, y), _m)),
```

Comment:
it might be nice to pull these out as `scaleMatrix`, `shearMatrix` etc constructors.

Reply:
totally! I think a lot of things could be optimized/added. might be fun to add a benchmark suite to see what speed improvements are needed.